electron/commands/paste-image.ts

```diff
@@ -0,0 +1,63 @@
+import { getMainWindow } from "../main-window";
+import logit from "../utils/logit";
+import fs from "fs";
+import { tmpdir, homedir } from "os";
+import path from "path";
+import { ELECTRON_COMMANDS } from "../../common/electron-commands";
+
+interface IClipboardFileParameters {
+  name: string;
+  extension: string;
+  size: number;
+  type: string;
+  encodedBuffer: string;
+}
+
+const createTempFileFromClipboard = (
+  appFolderPrefix: string,
+  inputFileParams: IClipboardFileParameters,
+  onSuccessCallback: (inputFilePath: string, outputFilePath: string) => void,
+  onErrorCallback: (error: Error) => void,
+) => {
+  let tempDirectory = fs.mkdtempSync(path.join(tmpdir(), appFolderPrefix));
+  let tempFilePath = path.join(tempDirectory, inputFileParams.name);
+
+  fs.writeFile(
+    tempFilePath,
+    Buffer.from(inputFileParams.encodedBuffer, "base64"),
+    (err) => {
+      if (err) {
+        onErrorCallback(new Error("No permission to temp folder"));
+        return;
+      }
+      let homeDirectoryAsOutputFolder = homedir();
+      onSuccessCallback(tempFilePath, homeDirectoryAsOutputFolder);
+    },
+  );
+};
+
+const pasteImage = (event, file: IClipboardFileParameters) => {
+  const mainWindow = getMainWindow();
+  if (!mainWindow) return;
+  if (!file || !file.name || !file.encodedBuffer) return;
+  const appFolderPrefix = "upscayl";
+  createTempFileFromClipboard(
+    appFolderPrefix,
+    file,
+    (imageFilePath, homeDirectory) => {
+      mainWindow.webContents.send(ELECTRON_COMMANDS.PASTE_IMAGE_SAVE_SUCCESS, [
+        imageFilePath,
+        homeDirectory,
+      ]);
```

Comment:
@NayamAmarshe @aaronliu0130 
what is your opinion?

Reply:
> @NayamAmarshe @aaronliu0130 what is your opinion? Both frontend and backend are in the same app, so I thought it is not an issue.. or should we let the user choose the output directory instead?

What are we using home directory for? Most packages won't allow such unrestricted access to home directory unless specified by the user. We should only be using the output directory to save the temporary image.

Reply:
The temporary directory of the OS is the input directory and home directory of the OS is the output directory as default. 
That was what I thought would be the best implementation. Both of these folders will be accessible by current user. The path module has tempdir() and homedir() functions to facilitate this. Otherwise I can remove home directory as the output and let the user choose

Reply:
Implementation method changed..
Now, output folder itself will be used as the temporary folder and if not given it will raise a toast..